src/functions.date.js

```diff
@@ -108,28 +108,56 @@ for (let unit in s) {
 	}, {multiValued: true});
 }
 
-_.duration = $.extend(function($this, ms) {
+_.duration = $.extend(function ($this, ms, terms) {
 	if (arguments.length === 1) {
 		[ms, $this] = [$this, null];
 	}
 
-	var count = ms || 0;
-	var unit = "ms";
+	if (ms===0 || terms === undefined) {
+		terms = 1;
+	}
+
+	let timeLeft = ms || 0;
+	let ret = [];
+	let unitsToUse = ["ms"];
 
+	// populate unitsToUse with all the terms that can fit within timeLeft
+	// in order of lowest to highest magnitude,
+	// so we have a list of consecutive unit terms to later reference
 	for (let nextUnit in s) {
-		var nextCount = _.msTo(nextUnit, ms);
+		let count = _.msTo(nextUnit, timeLeft);
```

Comment:
I think there's a bug here. `timeLeft` sounds like it would be updated in every iteration but it's not, so essentially you are always converting `ms`?

Since you are always converting the same number, you may have something that converts to an integer number for a given unit but once that is subtracted, it would be 0.
E.g. `_.msTo("days", 24 * 60 * 60 * 1000)` returns 1, but `_.msTo("hours", 24 * 60 * 60 * 1000)` returns 24, so it would be included.

I think you don't need two loops, you can just get all candidate units via `Object.keys(s)` and loop over that.

Also, I'd call that just `units`. Of course they are "to use", otherwise why are we defining them?